ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/parser/TestDirective.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.parser;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Functions;
+import io.confluent.ksql.parser.ParsingException;
+import io.confluent.ksql.parser.SqlBaseLexer;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.antlr.v4.runtime.Token;
+
+/**
+ * A test directive indicates special handling of test-only metadata. The
+ * directive is specified in the test sql file as a single-line comment that
+ * begins with * {@code --@} and the directive type and contents are separated
+ * by a colon. The type must contain only letters and "." characters.
+ *
+ * <p>For example, tests are delimited using the {@code TEST} directive, which
+ * would be specified in a file using {@code --@test: name}. The directive types
+ * are case-insensitive.</p>
+ *
+ * <p>The ANTLR grammar that parses comments will handle directives differently
+ * from normal comments, and will send them to the {@link SqlBaseLexer#DIRECTIVES}
+ * channel. See {@link SqlTestReader} for more handling of directives.</p>
+ */
+public class TestDirective {
+
+  static final Pattern DIRECTIVE_REGEX = Pattern.compile(
+      "--@(?<type>[a-zA-Z.]+):\\s*(?<contents>.*)"
+  );
+
+  private final Type type;
+  private final String contents;
+
+  public static TestDirective parse(final Token comment) {
+    final Matcher matcher = DIRECTIVE_REGEX.matcher(comment.getText().trim());
+    if (!matcher.find()) {
+      throw new ParsingException(
+          "Expected directive matching pattern " + DIRECTIVE_REGEX + " but got " + comment,
+          null,
+          comment.getLine(),
+          comment.getCharPositionInLine()
+      );
+    }
+
+    final Type type = Type.from(matcher.group("type").toLowerCase());
+    final String contents = matcher.group("contents");
+
+    return new TestDirective(type, contents);
+  }
```

Comment:
I'd move parsing out into own `Parser` class and leave `TestDirective` as a POJO.